### PR TITLE
Add API endpoint that allows creating a Keyspace

### PR DESCRIPTION
**Pull request by nastra**

### CODE REVIEW

File: management-api-agent/src/main/java/com/datastax/mgmtapi/NodeOpsProvider.java [hunk 1]
```diff
@@ -318,4 +321,18 @@ public void createRole(@RpcParam(name="username") String username,
     {
         return ShimLoader.instance.get().getStreamInfo();
     }
+
+    @Rpc(name = "createKeyspace")
+    public void createKeyspace(String keyspaceName, Map<String, String> replicationSettings) throws IOException
+    {
+        logger.debug("Creating keyspace {} with replication settings {}", keyspaceName, replicationSettings);
+        Map<String, Integer> settings = new HashMap<>();
```
Comment (RazakDangi) [inline management-api-agent/src/main/java/com/datastax/mgmtapi/NodeOpsProvider.java:329]: [minor] using map(transforming) would stop us from creating new local map with new
		
Map<String, Integer> settings = replicationSettings.entrySet().stream()
				.collect(
						Collectors.toMap(Map.Entry::getKey, entry ->Integer.valueOf(entry.getValue()))
				);

File: management-api-agent/pom.xml [hunk 2]
```diff
@@ -69,6 +69,11 @@
                     <version>4.11</version>
                     <scope>test</scope>
                 </dependency>
+                <dependency>
```
Comment (RazakDangi) [inline management-api-agent/pom.xml:72]: [minor] I would keep compile dependencies above test scope dependencies

Comment (nastra) [reply management-api-agent/pom.xml:72]: fixed

File: management-api-agent/src/main/java/com/datastax/mgmtapi/NodeOpsProvider.java (hunk 1 above)
Comment (nastra) [reply management-api-agent/src/main/java/com/datastax/mgmtapi/NodeOpsProvider.java:329]: I was only checking something with this code, but yes, the replication settings will usually always be a pair of String, Integer

**Comment by nastra** [comment]
> I think the operator also needs an alter keyspace replication for when it brings up a new DC.

@tjake np I can add that. Maybe as a separate issue/PR?

**Pull request merged**